services/src/main/java/org/keycloak/truststore/FileTruststoreProviderFactory.java

```diff
@@ -106,24 +112,94 @@ public void init(Config.Scope config) {
             }
         }
 
+        if (includeDefault) {
+            truststore = ensureTrustStoreIsWritable(storepath, truststore);
+            File defaultTrustStore = getDefaultTrustStoreFile();
+            if (defaultTrustStore.exists()) {
+                String trustStoreType = System.getProperty("javax.net.ssl.trustStoreType", KeyStore.getDefaultType());
+                String path = defaultTrustStore.getAbsolutePath();
+                mergeTrustStore(truststore, path, loadStore(path, trustStoreType, null));
+            } else {
+                log.warnf("Default truststore was to be included, but could not be found at: %s", defaultTrustStore, type);
+            }
+        }
+
+        if (additionalFiles != null && additionalFiles.length > 0) {
+            truststore = ensureTrustStoreIsWritable(storepath, truststore);
+            for (String file : additionalFiles) {
+                if (file.endsWith(".p12") || file.endsWith(".pfx")) {
```

Comment:
Also the `.pkcs12` should be taken into consideration, right?

Reply:
@mabartos it can if that's a common extension, all I could find a reference for were those.

Reply:
@shawkins Ahh, it seems only those two are common extensions. I was a little bit confused by the `.pkcs12` files in the Keycloak testsuite. Leave it as is.